lib/SILOptimizer/Utils/KeyPathProjector.cpp

```diff
@@ -230,7 +230,12 @@ class GettablePropertyProjector : public ComponentProjector {
       auto addr = builder.createAllocStack(loc, type);
       
       assertHasNoContext();
-      assert(getter->getArguments().size() == 2);
+      // For wasm, SILGen emits keypath projector with extra generic param
+      // argument to match callee and caller signature evens if it doesn't
+      // have generic params. WebAssembly *requires* callee and caller signature
+      // to always match.
+      auto target = builder.getASTContext().LangOpts.Target;
```

Comment:
I assume this would trigger an "unused variable" warning in an no-assert build.
Can you inline 'target' into the assert expression?

Reply:
I think on non-WebAssembly platforms it should be covered by existing test cases, while for WebAssembly itself there's no CI set up yet, so even if I add a test case here, it won't be run by @swift-ci.

Reply:
You might be able to get enough of it standalone to run the test without the standard library, which would let you test this in CI.

Reply:
Do you mean merging the rest of the toolchain changes without the stdlib changes? That would still require merging those toolchain PRs and setting up a community CI for WASI before this one is merged, right?

Reply:
This needs a test case and not only that, the optimizer does not generally assume specific targets like this. We need to find a way to encode this without referring to the target.

My suggestion is to write a SIL test that passes to sil-opt that this is the stdlib so it is self contained. You can do this by passing in the flags -parse-stdlib -module-name Swift.

Please /never/ add changes to the optimizer without tests.

Reply:
What would be the best way to encode this without referring to the target in your opinion? Also, is setting up a community CI for this target and enabling toolchain and build system support for WebAssembly and WASI are prerequisites for this test?

Reply:
No it is not a pre-requisite for writing this test.

Reply:
What is web assembly doing something different here.